pkgs/development/tcl-modules/by-name/ji/jimlib/package.nix

```
@@ -0,0 +1,45 @@
+{
+  lib,
+  fetchFromGitLab,
+  mkTclDerivation,
+}:
+
+mkTclDerivation rec {
+  pname = "jimlib";
+  version = "0.16.0";
+
+  src = fetchFromGitLab {
+    owner = "dbohdan";
+    repo = "jimlib";
+    rev = "v${version}";
+    sha256 = "sha256-a7Su/JByGLgnYBCRbP9z9CunTIYTwAisWuQafZWYgYo=";
+  };
+
+  dontBuild = true;
+  doCheck = true;
+
+  checkPhase = ''
+    runHook preCheck
+    tclsh ./jimlib.tcl
+    runHook postCheck
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm644 jimlib.tcl $out/lib/jimlib/jimlib.tcl
+    install -Dm644 pkgIndex.tcl $out/lib/jimlib/pkgIndex.tcl
+    runHook postInstall
+  '';
+
+  meta = {
+    homepage = "https://gitlab.com/dbohdan/jimlib/";
+    description = "Personal standard library for Jim Tcl.";
```

A detail I forgot on the previous round:

```suggestion
    description = "Personal standard library for Jim Tcl";
```

This is all documented in https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes by the way